src/pip/_internal/index/collector.py

```diff
@@ -38,10 +39,29 @@
     HTMLElement = xml.etree.ElementTree.Element
     ResponseHeaders = MutableMapping[str, str]
 
+    # Used in the @lru_cache polyfill.
+    F = TypeVar('F')
+
+    class LruCache(Protocol):
+        def __call__(self, maxsize=None):
+            # type: (Optional[int]) -> Callable[[F], F]
+            raise NotImplementedError
+
 
 logger = logging.getLogger(__name__)
 
 
+def noop_lru_cache(maxsize=None):
+    # type: (Optional[int]) -> Callable[[F], F]
+    def _wrapper(f):
+        # type: (F) -> F
+        return f
+    return _wrapper
+
+
```

Comment:
A comment mentioning Python 2 ("Fallback to noop_lru_cache in Python 2") would increase the chances we'll clean this when dropping python 2 support

Reply:
Yes, absolutely. This was lost when I refactored it. Will add!